### ISSUE TITLE
Scan barcode with flashlight on

### ISSUE THREAD

**Issue by rafwell** [phonegap/phonegap-plugin-barcodescanner#359]
How can i scan a barcode with flashlight on? I'm using cordova (ionic) and tried this code, but the flashlitgh switch off when barcode show the camera.  :(

I know is possible start flashlight pressing volume up key, but i wanna be this (flashlight on) the default behavior. Thanks.

```
$cordovaFlashlight.switchOn();

            $cordovaBarcodeScanner
            .scan()
            .then(function(barcodeData) {                
                // Success! Barcode data is here
                $log.info('Código de barras escaneado: '+barcodeData.text);
                
                }).catch(function(err){
                    $log.error(err);
                });

            }, function(error) {
                // An error occurred                
                $log.error('Houve um erro ao ler o código de barras. Por favor, tente novamente. '+error);
            });
```

**Comment by vespino** [comment]
The current plugin does support the flashlight, but only by enabling this using the volume up button. I have asked @EddyVerbruggen (one of the developers of this plugin) to add this feature. He will look into this for both Android and iOS.

**Comment by EddyVerbruggen** [comment]
Thanks to a kind bit of sponsorship by @vespino I was able to add this feature today. Available in 6.0.4 when passing the `showTorchButton : true` option to the `scan` function.
